### ISSUE TITLE
Make it easier to `String.slice` to the end of a string

### ISSUE THREAD

**Issue by spotandjake** [grain-lang/grain#1566]
currently if i wanted to do something similar to `"test".slice(1)` in js to get `"est"`, In Grain the only way to currently do this would be `String.slice(1, String.length("test"), "test") == "est"` Negative indexing feels like a fix to this but `String.slice(1, -1, "test") == "es"`, My solutiin to this was `String.slice(0, -0, "test") == "est"` but after asking @ospencer about this in the discord it was noted that we should never ever use `-0`, One possible solution to this would be to make `String.slice` use ranges then we could have inclusive or exclusive ranges but this is somewhat blocked by #1404 and would be a breaking change, another possible nonbreaking solution would be adding a `String.sliceIndex(<splitIndex>, "string")` but that would be less flexible and might get confusing.

TLDR: currently slicing from an index to the end of a string is extremly verbose, it would be nice if it was made simplier.

**Comment by alex-snezhko** [comment]
If ranges were to be used instead then wouldn't you still have the same problem of not being able to use a shortcut? For example you'd still have to call `String.length` for the upper bound of the range

**Comment by spotandjake** [comment]
Grain has `Inclusive` And `Exclusive` ranges. so you could use an inclusive range for that behaviour.

**Comment by alex-snezhko** [comment]
Yes but you'd still have to call `String.length` wouldn't you? It'd either be `Range.Inclusive(0, String.length(s) - 1)` or `Range.Exclusive(0, String.length(s))`

**Comment by ospencer** [comment]
Yeah, we haven't really talked about negative indexing on ranges.

**Comment by alex-snezhko** [comment]
I suppose a hacky alternative would be to allow `Infinity` as the end index although that doesn't seem like a great compromise

**Comment by phated** [comment]
This isn't a problem once we have optional arguments, so I'd vote that we don't "solve" it until then. Once optional arguments exist, the "end" argument will be optional
